### DISCUSSION TITLE
Zwei kleine Verbesserungsvorschlaege

### DISCUSSION THREAD

**Discussion by esusxunil** [solectrus/solectrus#2402] [Ideas]
Erst einmal vielen Dank fuer das grossartige Tool und die tolle Arbeit. Es ist um soviele Laengen besser, als die Tools direkt vom Hersteller.

Ich haette zwei kleine Anregungen:

- Anzeige Leistungsbegrenzung: Ich weis es ist nicht mehr so viele Interessant, da viele die schon rausnehmen durften - unsere Anlage ist jedoch zu gross und so sind wir noch auf 70% Einspeiseleistung vom Wechselrichter gedrosselt. Der einzige Ort in dem das ersichtlich ist, ist die lokale Webseite auf dem Geraet. Bei uns funktioniert das Peak Shaving ganz gut, somit laufen wir da nicht mehr so oft rein, aber wenn, dann sehen wir es nur dort. Eine Anzeige der Drosselung (wenn gerade Gedrosselt) waere grossartig.

- Die Anzeige der Ersparnis bezieht sich ja nur darauf was man spart im Vergleich zum Strom aus dem Netz. Wie ware noch eine weiter Anzeige der Kosten, wenn man die geladenen KW im Auto per Treibstoff verfahren haette?

Vielen Dank und mache bitte weiter so.

**Comment by ledermann** [comment]
@esusxunil: Freut mich, dass dir SOLECTRUS gefällt :)

Zu deinen Anregungen:

## Leistungsbegrenzung

Bei mir ist die Einspeisebegrenzung auch aktiv und im Sommer greift das auch, was man dann im Kurvenverlauf sehen kann (z.B. hier: https://solectrus.ledermann.dev/grid_power/2023-09-06). 

Ja, es wäre interessant zu wissen, wenn gerade auf 70 % gedrosselt wird. Aber wo genau siehst du das denn auf der lokalen Website des Speichers? Hast du einen Screenshot oder genauere Informationen dazu?

## Verbrenner-Vergleich

Ein solcher Vergleich würde einige zusätzliche Angaben benötigen: 
  - Verbrauch eines fiktiven Verbrenners in L/100km
  - Spritpreisentwicklung im Zeitverlauf (notfalls ein konstanter Wert)
  - Verbrauch des genutzten E-Autos in kWh/100 km

Außerdem: Was ist mit Ladungen an Ladesäulen, die gar nicht erfasst werden?
  
Die Berechnung ist vermutlich nicht ganz trivial, denn: Würde man einen Verbrenner fahren, würde man weniger Strom verbrauchen und somit weniger Strom einkaufen und mehr einspeisen. Die Ersparnis-Berechnung der PV-Anlage wird also beeinflusst.

Vielleicht genügt aber auch eine ganz einfache Darstellung wie folgt: "Du hast diesen Monat 134 kWh ins E-Auto geladen. Damit konntest du 1.030 km weit fahren (wegen 13 kWh/100km). Ein Verbrenner hätte für die gleiche Strecke 72 L Diesel verbraucht (wegen 7L/100km) und dich 130 € gekostet (wegen 1,80€/L)." 

Also ganz ohne Ersparnis-Berechnung. Was meinst du dazu?

**Comment by esusxunil** [comment]
### Leistungsbegrenzung

Auf der Lokalen Webseite unter "Information" im Bereich Leistungswerte unten links sehe ich die Leistungsbegrenzung der drei Phasen

![image](https://github.com/solectrus/solectrus/assets/61476952/99452dce-d94a-453f-a45b-267e8270ce7d)

Aktuell (spaet abends) dreimal 100% - zur MIttagszeit wenn die Anlage brummt - faellt der Wert ab.

### Verbrenner-Vergleich

Die zusaetlichen Angaben erscheinen mir plausibel. Moeglich waere auch ein fester Umrechnungsfaktor. Auch laedt man ja evtl. mehr als ein Auto, dass es am Ende nur ein Naeherungswert sein kann ist mir klar.

Spritpreise habe ich mir vor dem Umstieg auf Elektro ueber Tankerkoenig geholt.

Ich denke Ladungen ausserhalb kann man Vernachlaessigen, da der Focus ja immer noch auf dem Ertrag/den Einsparungen der PV liegt, nicht die vom Auto.

Das mit dem Einfluss auf den Ertrag hatte ich nicht bedacht. Aktuell vergleicht du ja eigentlich zwei Szenarien miteinander (mIt/ohne PV), fliest der Verbrenner Vergleich damit ein, hat man zwei Faktoren, die sich gegenseitig beeinflussen, also vier Szenarien (mit pv + Eauto, mit pv + Verbrenner, ohne Pv+ Eauto, ohne Pv + Verbrenner).

Die einfache Darstellung gefaellt mir sehr gut, wuerde mir sehr gefallen.

**Reply by ledermann** [reply]
Alles klar, Darstellung der Leistungsbegrenzung macht Sinn, ich habe das als Issue #2406 angelegt und bereits implementiert, weil es so einfach ist. Wird im nächsten Update enthalten sein.

Der Verbrenner-Vergleich ist auch spannend, aber deutlich aufwändiger zu implementieren. Ich habe das in #2407 schon mal als Issue angelegt.

**Comment by StevieC121176** [comment]
Sehr spannendes Thema. So eine Berechnung nutze ich auch momentan über iobroker. Es wäre super wenn das auch über Solectrus möglich wäre. 
Zum Laden des E-Autos habe ich zusätzlich noch EVCC, damit ist es im Großteil des Jahres möglich rein mit PV Energie zu laden ohne Netzbezug. Praktischerweise können die Ladevorgänge in eine *.csv gespeichert werden. Mit dem Verbrauch des Autos und dem Vergleichsverbrauch sowie aktuellem Spritpreis (per API abrufbar) kann man so sehr gut eine Vergleichsrechnung machen.
 
[session-2023-09.csv](https://github.com/solectrus/solectrus/files/12761801/session-2023-09.csv)

**Reply by ledermann** [reply]
Ja, wenn kein SENEC-Speicher bzw. keine Original-SENEC-Wallbox vorliegt, muss ein Weg gefunden werden, wie die Ladevorgänge in die InfluxDB kommen. Das wird nochmal ein größeres Thema ;-)
